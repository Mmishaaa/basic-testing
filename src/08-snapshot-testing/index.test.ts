// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const values = {
      value: 10,
      next: {
        value: 11,
        next: {
          value: 12,
          next: {
            value: 13,
            next: { value: 100, next: { value: null, next: null } },
          },
        },
      },
    };
    expect(values).toStrictEqual(generateLinkedList([10, 11, 12, 13, 100]));
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const values = {
      value: 10,
      next: {
        value: 11,
        next: {
          value: 12,
          next: {
            value: 13,
            next: { value: 100, next: { value: null, next: null } },
          },
        },
      },
    };
    expect(values).toMatchSnapshot(generateLinkedList([10, 11, 12, 13, 100]));
  });
});
