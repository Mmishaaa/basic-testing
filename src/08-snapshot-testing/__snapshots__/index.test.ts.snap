// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": {
            "next": null,
            "value": null,
          },
          "value": 100,
        },
        "value": 13,
      },
      "value": 12,
    },
    "value": 11,
  },
  "value": 10,
}
`;
